corehq/apps/aggregate_ucrs/column_specs.py

```diff
@@ -0,0 +1,335 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import sqlalchemy
+from django.utils.translation import ugettext_lazy as _
+from abc import ABCMeta, abstractmethod, abstractproperty
+
+import six
+from jsonobject.base_properties import DefaultProperty
+
+from corehq.apps.aggregate_ucrs.aggregations import AGG_WINDOW_START_PARAM
+from corehq.apps.aggregate_ucrs.query_column_providers import StandardQueryColumnProvider, \
+    AggregationParamQueryColumnProvider
+from corehq.apps.userreports.datatypes import DATA_TYPE_INTEGER, DataTypeProperty, DATA_TYPE_STRING, DATA_TYPE_DATE
+from corehq.apps.userreports.indicators import Column
+from dimagi.ext import jsonobject
+
+
+class ColumnAdapater(six.with_metaclass(ABCMeta, object)):
+    """
+    A column adapter represents everything needed to work with an aggregate column,
+    including its config as well as its sqlalchemy information (via the ucr column spec).
+
+    Column adapters handle both the querying side (via the to_sqlalchemy_query_column function)
+    as well as the table creation/writing side.
+    """
+    config_spec = jsonobject.JsonObject
```

Comment:
Should you make this abstract so that subclasses have to override it?

Reply:
am not sure. `RawColumnAdapter` doesn't use/override it at the moment so maybe keeping a default is ok? I added a comment to make it clear that it's usually overridden, though not sure that helps